src/viadot/sources/smb.py

```diff
@@ -0,0 +1,330 @@
+
+        Returns:
+            Iterator: An iterator of directory entries.
+        """
+        return smbclient.scandir(path)
+
+    def _is_matching_file(
+        self,
+        entry: smbclient._os.SMBDirEntry,
+        keywords: list[str] | None = None,
+        extensions: list[str] | None = None,
+        date_filter_parsed: pendulum.Date
+        | tuple[pendulum.Date, pendulum.Date]
+        | None = None,
+    ) -> bool:
+        """Check if a file matches the given criteria.
+
+        It verifies whether the file satisfies any combination of:
+        - Keyword-based filtering.
+        - Extension-based filtering.
+        - Exact date or date range filtering.
+
+        Args:
+            entry (smbclient._os.SMBDirEntry): A directory entry object from
+                the directory scan.
+            keywords (list[str] | None): List of keywords to search for in filenames.
+                It is case-insensitive. Defaults to None.
+            extensions (list[str] | None): List of file extensions to filter by.
+                It is case-insensitive. Defaults to None.
+            date_filter_parsed (
+                pendulum.Date | tuple[pendulum.Date, pendulum.Date] | None
+            ):
+                - A single `pendulum.Date` for exact date filtering.
+                - A tuple of two `pendulum.Date` values for date range filtering.
+                - None, if no date filter is applied.
+                Defaults to None.
+
+        Returns:
+            bool: True if the file matches all criteria, False otherwise.
+        """
+        name_lower = entry.name.lower()
+
+        matches_extension = not extensions or any(
+            name_lower.endswith(ext.lower()) for ext in extensions
+        )
+        matches_keyword = not keywords or any(
+            keyword.lower() in name_lower for keyword in keywords
+        )
+        if not matches_extension or not matches_keyword or entry.is_dir():
+            return False
+
+        if date_filter_parsed:
+            file_creation_date = pendulum.from_timestamp(entry.stat().st_ctime).date()
+            if isinstance(date_filter_parsed, pendulum.Date):
+                return file_creation_date == date_filter_parsed
+            if isinstance(date_filter_parsed, tuple):
+                start_date, end_date = date_filter_parsed
+                return start_date <= file_creation_date <= end_date
+
+        return True
+
+    ## Remove this function because its no needed
+    # def _store_matching_file(self, file_path: str) -> None:
+    #     """Process a matching file by fetching its content.
+
+    #     It fetching the content and storing it in the found_files dictionary.
+
+    #     Args:
+    #         file_path (str): The full path of the matching file.
+    #     """
+    #     self.logger.info(f"Found: {file_path}")
+    #     content = self._fetch_file_content(file_path)
+    #     self.found_files[file_path] = content
```

Comment:
Remove this if it's no longer required pls

Reply:
[9347998](https://github.com/dyvenia/viadot/pull/1127/commits/93479985e3e87b179231c0581796e71332649e83)